src/workers/notifications/postAdded.ts

```diff
@@ -61,13 +65,20 @@ const worker: NotificationWorker = {
       }
       if (source.type === SourceType.Squad && post.authorId) {
         // squad_post_added notification
-        const doneBy = await con
-          .getRepository(User)
-          .findOneBy({ id: post.authorId });
+        const [doneBy, mutes] = await Promise.all([
+          con.getRepository(User).findOneBy({ id: post.authorId }),
+          con.getRepository(NotificationPreferenceSource).findBy({
+            referenceId: source.id,
+            status: NotificationPreferenceStatus.Muted,
+            notificationType: NotificationType.SquadPostAdded,
+          }),
+        ]);
+        const ignored = mutes.map(({ userId }) => userId);
+        ignored.push(post.authorId);
         const members = await con.getRepository(SourceMember).find({
           where: {
             sourceId: source.id,
-            userId: Not(In([post.authorId])),
+            userId: Not(In(ignored)),
```

Comment:
I suggest using a subquery here instead of fetching the users and then providing them back to the database.
It's much more scalable and fast